internal/util/search/search.go

```diff
@@ -83,26 +90,46 @@ func (sr *SearchReplace) Perform(resourcesPath string) error {
 
 	return kio.Pipeline{
 		Inputs:  []kio.Reader{inout},
-		Filters: []kio.Filter{kio.FilterAll(sr)},
+		Filters: []kio.Filter{filterAll(sr)},
 		Outputs: []kio.Writer{inout},
 	}.Execute()
 }
 
 // Filter parses input node and performs search and replace operation on the node
 func (sr *SearchReplace) Filter(object *yaml.RNode) (*yaml.RNode, error) {
+	originalObject, err := deepCopy(object)
```

Comment:
What should be the level at which we roll back changes if validation fails? It seems like we do in on a per-resource basis here, but it's not obvious that we shouldn't handle this on a per-package or package+subpackages basis.